app/views/main/_practice_area.html.erb

```diff
@@ -2,27 +2,19 @@
 <br />
 <p>Тук можете да се упражнявате на вече минали състезания.</p>
 <br />
-<% if !@practice_contests.blank? %>
-<%= will_paginate @practice_contests, :param_name => :practice_contests_page %>
-<% cache ["practice_area", params[:practice_contests_page] || 1, Contest.last, Run.where(:user_id => current_user.id).last] do %>
-  <table>
-    <tr>
-      <th>Име</th>
-      <th>Вашите точки</th>
-      <th></th>
-    </tr>
-    <% @practice_contests.each do |contest| %>
+<% if !@contest_groups.blank? %>
+    <%= will_paginate @contest_groups, :param_name => :contest_groups_page %>
+    <table>
       <tr>
-        <td><%= contest.name %></td>
-        <td><strong><%= current_user.best_practice_score(contest) %></strong> от максимално <%= contest.max_score %></td>
-        <td>
-          <%= link_to "Практикувай", open_contest_practice_path(:contest_id => contest.id) %>
-        </td>
+        <th>Група</th>
       </tr>
-    <% end %>
-  </table>
-<% end %>
-<%= will_paginate @practice_contests, :param_name => :practice_contests_page %>
+      <% @contest_groups.each do |contest_group| %>
+          <tr>
+            <td><%= link_to contest_group.name, contest_group_path(contest_group) %></td>
+          </tr>
+      <% end %>
+    </table>
+    <%= will_paginate @contest_groups, :param_name => :contest_groups_page %>
 <% else %>
-В момента няма състезания, на които да можете да се упражнявате.
+    В момента няма състезания, на които да можете да се упражнявате.
```

Comment:
So, just to clarify that I understand correctly, we want on the main page both views. The view with the contests and the new view, which will be showing the groups, is that right ?